src/utils/responses.ts

```diff
@@ -0,0 +1,100 @@
+import {
+  HttpStatusCodeEnum,
+  ResponseErrorGeneric
+} from "@pagopa/ts-commons/lib/responses";
+import { PartyConfigurationFaultEnum } from "../../generated/api/PartyConfigurationFault";
+import { PartyTimeoutFaultEnum } from "../../generated/api/PartyTimeoutFault";
+import { PaymentFaultEnum } from "../../generated/api/PaymentFault";
+import { PaymentFaultV2Enum } from "../../generated/api/PaymentFaultV2";
+import { GatewayFaultEnum } from "../../generated/api/GatewayFault";
+import { ValidationFaultEnum } from "../../generated/api/ValidationFault";
+import { PaymentStatusFaultEnum } from "../../generated/api/PaymentStatusFault";
+import {
+  IResponseGatewayError,
+  IResponseGatewayTimeout,
+  IResponsePartyConfigurationError,
+  IResponseErrorValidationFault,
+  ResponseGatewayTimeout,
+  ResponsePartyConfigurationError,
+  ResponsePaymentError,
+  IResponsePaymentStatusFaultError,
+  ResponsePaymentStatusFaultError
+} from "./types";
+import { logger } from "./Logger";
+
+export const ResponseErrorValidationFault: (
+  title: string,
+  detail: ValidationFaultEnum
+) => IResponseErrorValidationFault = (title, detail) => {
+  // eslint-disable-next-line functional/no-let
+  let responseDetail;
+  if (typeof detail === "string") {
+    responseDetail = `${title}: ${detail}`;
+  } else {
+    responseDetail = detail;
+  }
+  return {
+    ...ResponseErrorGeneric(HttpStatusCodeEnum.HTTP_STATUS_404, title, detail),
+    detail: responseDetail,
+    kind: "IResponseErrorValidationFault"
+  };
+};
+
+export const responseFromPaymentFault: (
+  detail: PaymentFaultEnum,
+  detail_v2: PaymentFaultV2Enum
+) =>
+  | IResponsePaymentStatusFaultError
+  | IResponsePartyConfigurationError
+  | IResponseErrorValidationFault
+  | IResponseGatewayError
+  | IResponseGatewayTimeout = (detail, detail_v2) => {
+  if (
+    Object.values(PartyConfigurationFaultEnum).includes(
```

Comment:
what do you think if you refactor with `decode` rather than object / if?

Reply:
I think I managed to get a more elegant solution in 8894417, thanks for the heads up 😃